Source/WebCore/platform/graphics/gstreamer/VideoFrameGStreamer.cpp

```diff
@@ -30,6 +30,16 @@
 
 namespace WebCore {
 
+static inline void setBufferFields(GstBuffer* buffer, const MediaTime& presentationTime, double frameRate)
+{
+    GST_BUFFER_FLAG_SET(buffer, GST_BUFFER_FLAG_LIVE);
+
+    GST_BUFFER_DTS(buffer) = GST_BUFFER_PTS(buffer) = toGstClockTime(presentationTime);
+
+    auto duration = 1_s / frameRate;
+    GST_BUFFER_DURATION(buffer) = toGstClockTime(MediaTime::createWithDouble(duration.seconds()));
```

Comment:
I've seen you using this pattern a couple of times already. You might want to overload `toGstClockTime` to take `const Seconds &`

Reply:
Good idea!